conda/exceptions.py

```diff
@@ -590,36 +593,170 @@ def print_conda_exception(exception):
+
+    return do_upload, ask_for_upload, stdin
+
+
+def _execute_upload(context, error_report):
+    headers = {
+        'User-Agent': context.user_agent,
+    }
+    _timeout = context.remote_connect_timeout_secs, context.remote_read_timeout_secs
+    data = json.dumps(error_report, sort_keys=True, cls=EntityEncoder) + '\n'
+    try:
+        # requests does not follow HTTP standards for redirects of non-GET methods
+        # That is, when following a 301 or 302, it turns a POST into a GET.
+        # And no way to disable.  WTF
+        import requests
+        q = 0
+        url = context.error_upload_url
+        response = requests.post(url, headers=headers, timeout=_timeout, data=data)
+        response.raise_for_status()
+        while response.status_code in (301, 302) and response.headers.get('Location'):
+            url = response.headers['Location']
+            response = requests.post(url, headers=headers, timeout=_timeout, data=data)
+            response.raise_for_status()
+            q += 1
+            if q > 15:
+                raise CondaError("Redirect limit exceeded")
+        log.debug("upload response status: %s", response and response.status_code)
+    except Exception as e:  # pragma: no cover
+        log.info('%r', e)
 
-    https://github.com/conda/conda/issues
 
-"""
-        stderrlogger.info(message)
-        command = ' '.join(ensure_text_type(s) for s in sys.argv)
-        if ' info' not in command:
-            from .cli.main_info import get_info_dict, get_main_info_str
-            stderrlogger.info(get_main_info_str(get_info_dict()))
-        stderrlogger.info("`$ {0}`".format(command))
-        stderrlogger.info('\n')
-        stderrlogger.info('\n'.join('    ' + line for line in traceback.splitlines()))
+def print_unexpected_error_message(e):
+    try:
+        traceback = format_exc()
+    except AttributeError:  # pragma: no cover
+        if sys.version_info[:2] == (3, 4):
+            # AttributeError: 'NoneType' object has no attribute '__context__'
+            traceback = ''
+        else:
+            raise
+
+    from .base.context import context
+
+    command = ' '.join(ensure_text_type(s) for s in sys.argv)
+    info_dict = {}
+    if ' info' not in command:
+        try:
+            from .cli.main_info import get_info_dict
+            info_dict = get_info_dict()
+        except Exception as info_e:
+            info_traceback = format_exc()
+            info_dict = {
+                'error': repr(info_e),
+                'error_type': info_e.__class__.__name__,
+                'traceback': info_traceback,
+            }
+
+    error_report = {
+        'error': repr(e),
+        'error_type': e.__class__.__name__,
+        'command': command,
+        'traceback': traceback,
+        'conda_info': info_dict,
+    }
+
+    do_upload, ask_for_upload, stdin = _print_exception_message_and_prompt(context, error_report)
+
+    if do_upload:
+        _execute_upload(context, error_report)
```

Comment:
Overall this is a nice feature and should work fine when used in conjunction with `conda.cli.python_api` since there `context.always_yes` and thus `do_upload` will always be `True`.
Though for `run_command(..., use_exception_handler=True)` the user will only see
`"An unexpected error has occurred. Conda has prepared the following report.\n[report]"`
but no message about what has been done with that report.
It would be nice if after `_execute_upload` has been called, the status would be printed. Something like "Successfully uploaded X" or "Failed to upload X because of HTTP error code".